src/agentlab/agents/generic_agent/generic_agent.py

```diff
@@ -65,7 +66,7 @@ def __init__(
     def obs_preprocessor(self, obs: dict) -> dict:
         return self._obs_preprocessor(obs)
 
-    @openai_monitored_agent
+    @get_action_decorator
```

Comment:
maybe call it cost_tracker_decorator?

Reply:
should we put it as part of Agent base class? this way users won't have to add
the decorartor to track the cost

Reply:
```suggestion
class Agent:
   def _get_action(self,obs):
        with cost_tracker():
            return self.get_action(obs) 
```

Reply:
i'm not sure about this, would we define a new Agent class on top of bgym.Agent ?

Reply:
Ahhh crap, I didn't realized it was in browsergym. We could defined TrackedAgent, but hey let's leave it like that for now.